BrainPortal/lib/boutiques_input_subdir_maker.rb

```diff
@@ -0,0 +1,163 @@
+  # the fake parent directory information
+  def descriptor_for_form #:nodoc:
+    descriptor                = super.dup
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesInputSubdirMaker')
+
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      # Adjust the description
+      input              = descriptor.input_by_id(inputid)
+      dirname            = fake_parent_dirname[0]
+      input.description  = input.description.to_s +
+                           "\nThis input will be copied in a parent folder: #{dirname}. The parent folder will be used in the command line"
+    end
+
+    descriptor
+  end
+
+  ############################################
+  # Bourreau (Cluster) Side Modifications
+  ############################################
+
+  # For input in `BoutiquesInputSubdirMaker` section,
+  # create a fake parent directory that will contains a symlink
+  # to the orginal selected Userfile.
+  def setup #:nodoc:
+    original_userfile_ids = {}
+
+    # Log revision information
+    basename = Revision_info.basename
+    commit   = Revision_info.short_commit
+    self.addlog("Creating parent directories in BoutiquesInputSubdirMaker.")
+    self.addlog("#{basename} rev. #{commit}")
+
+    descriptor = self.descriptor_for_setup
+
+    # Remove IDs from invoke_params
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesInputSubdirMaker')
+    parent_dirname_by_inputid.keys do |inputid|
+      original_userfile_ids[inputid]  = invoke_params[inputid]
+      invoke_params[inputid]          = nil
+    end
+
+    # invoke main setup
+    result = super
+
+    # Return false if super failed
+    return false if ! super
+
+    # Special make_available who need to have a parent folder
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      dirname     = fake_parent_dirname[0]
+      userfile_id = original_userfile_ids[inputid]
+
+      next if userfile_id.blank?
+
+      userfile = Userfile.find(userfile_id)
+      make_available(userfile, "#{dirname}/#{userfile.name}")
+    end
+
+    true
+  ensure
+    original_userfile_ids.each do |inputid, userfile_id|
+      invoke_params[inputid] = original_userfile_ids[inputid]
+    end
+  end
+
+  # Overrides the same method in BoutiquesClusterTask, as used
+  # during cluster_commands()
+  def finalize_bosh_invoke_struct(invoke_struct) #:nodoc:
+    override_invoke_params = super.dup
+
+    descriptor = self.descriptor_for_cluster_commands
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesInputSubdirMaker')
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      if override_invoke_params[inputid].blank?
+        override_invoke_params.delete(inputid)
+      else
+        (dirname, append_userfile_name) = fake_parent_dirname
+        override_invoke_params[inputid] = append_userfile_name ? "#{dirname}/#{override_invoke_params[inputid]}" : dirname
```

Comment:
This doesn't work, you're going to set the entry in `override_invoke_params` to `dirname/userfileID` instead of `dirname/userfileNAME`

Reply:
Ok this works, the name has been substituted by the time this method is reached.